build/source/engine/groundwatr.f90

```diff
@@ -225,70 +209,7 @@ subroutine groundwatr(&
-
-   ! perturb state vector
-   mLayerMatricHeadPerturbed(iLayer) = mLayerMatricHeadPerturbed(iLayer) + dx
-
-   ! compute the columetruc liquid water content
-   mLayerVolFracLiqPerturbed(iLayer) = volFracLiq(mLayerMatricHeadPerturbed(iLayer),vGn_alpha(iLayer),theta_res(iLayer),theta_sat(iLayer),vGn_n(iLayer),vGn_m(iLayer))
-
-   ! compute baseflow flux
-   ! NOTE: This is an optional second call to computeBaseflow that is invoked when computing numerical derivatives.
-   !       Since the purpose here is to compute the numerical derivatives, we do not need to compute analytical derivatives also.
-   !       Hence, analytical derivatives are not desired
-   call computeBaseflow(&
-                        ! input: control and state variables
-                        nSnow,                     & ! intent(in): number of snow layers
-                        nSoil,                     & ! intent(in): number of soil layers
-                        nLayers,                   & ! intent(in): total number of layers
-                        .false.,                   & ! intent(in): .true. if analytical derivatives are desired
-                        ixSaturation,              & ! intent(in): index of upper-most "saturated" layer
-                        mLayerVolFracLiqPerturbed, & ! intent(in): volumetric fraction of liquid water in each soil layer (-)
-                        mLayerVolFracIce,          & ! intent(in): volumetric fraction of ice in each soil layer (-)
-                        ! input/output: data structures
-                        attr_data,                 & ! intent(in):    spatial attributes
-                        mpar_data,                 & ! intent(in):    model parameters
-                        prog_data,                 & ! intent(in):    model prognostic variables for a local HRU
-                        flux_data,                 & ! intent(inout): model fluxes for a local HRU
-                        ! output: fluxes and derivatives
-                        mLayerBaseflowPerturbed,   & ! intent(out): baseflow flux in each soil layer (m s-1)
-                        dBaseflow_dVolLiq)           ! intent(out): ** NOT USED ** derivative in baseflow w.r.t. volumetric liquid water content (s-1)
-
-   ! compute the numerical Jacobian
-   nJac(:,iLayer) = (mLayerBaseflowPerturbed(:) - mLayerBaseflow(:))/dx    ! compute the Jacobian
-
-   ! set the state back to the input value
-   mLayerMatricHeadPerturbed(iLayer) = mLayerMatricHeadLiq(iLayer)
-   mLayerVolFracLiqPerturbed(iLayer) = mLayerVolFracLiq(iLayer)
-
-  end do  ! looping through state variables
-
-  ! print the numerical Jacobian
-  print*, '** numerical Jacobian:'
-  write(*,'(a4,1x,100(i12,1x))') 'xCol', (iLayer, iLayer=1,nSoil)
-  do iLayer=1,nSoil; write(*,'(i4,1x,100(e12.5,1x))') iLayer, nJac(1:nSoil,iLayer); end do
-  !pause 'testing Jacobian'
-
- end if  ! if desire to compute the Jacobian
-
- ! end association to variables in data structures
- end associate
```

Comment:
Why can this section be taken out?

Reply:
It was my understanding that the numerical Jacobian was previously used to confirm the implementation of the analytic Jacobian (which is actually used in the models), and that the numerical Jacobian was no longer used. The fact that `doNumericalJacobian=.false.`  (line 147 in the original file) and cannot be changed without modifying the source code and recompiling is consistent with this. Perhaps @martynpclark can confirm?

Reply:
I've confirmed this with @martynpclark during a meeting on June 6th.